### PR TITLE
Update several requirements

### PR DESCRIPTION
**Pull request by mshriver**
I'm pulling in several package updates that were made by pyupbot on a commit with broken CI. Its easier than trying to update all of the bot's separate PRs.

### CODE REVIEW

**Review by JacobCallahan — APPROVED** [review]
Merge pending successful test run

**Comment by mshriver** [comment]
Thanks @JacobCallahan  - I'll have to remove one commit before we merge, that's pointing to my satellite6-upgrade branch.

Seeing odd gnutls vs openssl pycurl compilation failures in the GHA, which makes no sense as I haven't changed that environment at all.

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -28,7 +28,9 @@ jobs:
           sudo apt update
           sudo apt-get install -y libgnutls28-dev libcurl4-openssl-dev libssl-dev
           wget https://raw.githubusercontent.com/SatelliteQE/broker/master/broker_settings.yaml.example
-          pip install -U --no-binary=pycurl -r requirements.txt -r requirements-optional.txt
+          pip uninstall -y pycurl
+          pip install --compile --no-cache-dir pycurl
```
Comment (JacobCallahan) [inline .github/workflows/action.yml:32]: not blocking, but if for some reason you need to update this pr (other feedback) can you add a comment as to why these steps are necessary?

File: .github/workflows/update_robottelo_image.yml [hunk 2]
```diff
@@ -25,7 +25,9 @@ jobs:
         run: |
           sudo apt-get install -y libgnutls28-dev libcurl4-openssl-dev libssl-dev
           wget https://raw.githubusercontent.com/SatelliteQE/broker/master/broker_settings.yaml.example
-          pip install -U --no-binary=pycurl -r requirements.txt -r requirements-optional.txt
+          pip uninstall -y pycurl
+          pip install --compile --no-cache-dir pycurl
```
Comment (JacobCallahan) [inline .github/workflows/update_robottelo_image.yml:29]: same question

File: .github/workflows/action.yml (hunk 1 above)
Comment (mshriver) [reply .github/workflows/action.yml:32]: ack, added.

File: .github/workflows/update_robottelo_image.yml (hunk 2 above)
Comment (mshriver) [reply .github/workflows/update_robottelo_image.yml:29]: ack, added

**Review by devendra104 — APPROVED** [review]
LGTM :+1:

**Pull request merged**